### PR TITLE
#72 Implement  Two factor authentication for Vendor and Admin

### PR DESCRIPTION
**Pull request by Shumba250**
#### What does this PR do?

- Implement a two-factor authentication page for vendor login
#### Description of Task to be completed

- Implement a responsive two-factor authentication page
- Process OTP and redirect the vendor to the dashboard
- Validate the OTP input field
#### How should this be manually tested?

- clone the repo using git clone repo's URL
- change branch git checkout ft-two-factor-auth
- create .env file and paste variables from .env.example
- run npm install 
- run npm run dev
- signup a vendor and login
- Copy the OTP sent to your email and paste it in the input filed

### CODE REVIEW

**Comment by Shumba250** [comment]
[Seller auth & Two Factor Authentication for Sellers](https://trello.com/c/K58Vpe6U/72-seller-auth-two-factor-authentication-for-sellers)

**Review by sostene12 — COMMENTED** [review]
Screenshot 2023-05-01 at 14 49 06
this card is soo big, if you can reduce it

File: coverage/components/Auth/Input.jsx.html
Comment (sostene12) [inline coverage/components/Auth/Input.jsx.html:1]: i suggest you don't push coverage folder, please ignore it

File: src/pages/index.jsx
Comment (sostene12) [inline src/pages/index.jsx:1]: consider using app.jsx as the main file of routes

File: src/redux/actions/authAction.js
Comment (sostene12) [inline src/redux/actions/authAction.js:1]: it's not a good practice to use browser storage mean, if it is local storage stick with it only, or stick to using cookies only

**Review by sostene12 — CHANGES_REQUESTED** [review]

**Review by sostene12 — APPROVED** [review]

**Review by Luciefifi — APPROVED** [review]

**Review by chrill-christella — APPROVED** [review]

**Review by wakaflorien — APPROVED** [review]

**Pull request merged**